src/main/java/me/profelements/dynatech/items/electric/AntigravityBubble.java

```diff
@@ -57,48 +63,70 @@ public boolean isSynchronized() {
 
     @Override
     public void tick(Block b) {
+        Set<UUID> tempUuids = new HashSet<>();
+        tempUuids.clear();
         Collection<Entity> bubbledEntities = b.getWorld().getNearbyEntities(b.getLocation(), 25, 25, 25);
-
         for (Entity entity : bubbledEntities) {
             if (entity instanceof Player) {
                 Player p = (Player) entity;
-
+                if (allEnabledPlayers.get(b.getLocation()) == null) {
+                    tempUuids.add(p.getUniqueId());
+                    allEnabledPlayers.put(b.getLocation(), tempUuids);
+                } else {
+                    
+                    allEnabledPlayers.get(b.getLocation()).add(p.getUniqueId());
+                }
```

Comment:
Firstly this doesn't actually check if there is enough power to do the operation (pre-existing issue, not from this PR) so that needs sorting. And we can get rid of the temp uuid thing and grab the correct set of players outside of the iteration to save a bit of processing here.

Github wont let me do a proper change suggestion so here is what I think may be better (untested and not checked so please use with care :D)

```
    public void tick(Block block) {
        Collection<Entity> bubbledEntities = block.getWorld().getNearbyEntities(block.getLocation(), 25, 25, 25);
        Set<UUID> playersInBubble = allEnabledPlayers.getOrDefault(block.getLocation(), new HashSet<>());

        for (Entity entity : bubbledEntities) {
            if (entity instanceof Player && getCharge(block.getLocation()) >= getEnergyConsumption()) {
                Player player = (Player) entity;
                playersInBubble.add(player.getUniqueId());
                allEnabledPlayers.put(player.getLocation(), playersInBubble);
                if (!player.getAllowFlight()) {
                    player.setAllowFlight(true);
                    removeCharge(player.getLocation(), getEnergyConsumption());
                }
            }
        }

        for (UUID uuid : playersInBubble) {
            Player player = Bukkit.getPlayer(uuid);
            if (player != null && !bubbledEntities.contains(player)) {
                allEnabledPlayers.get(block.getLocation()).remove(player.getUniqueId());
                checkPlayer(player.getUniqueId());
            }
        }
    }
```

Reply:
I did not know this was possible:
`Set<UUID> playersInBubble = allEnabledPlayers.getOrDefault(block.getLocation(), new HashSet<>());`
I forgot to remove the clear on the tempUuids Set because I was trying out some different things.

And yeah I noticed the bubbles dont actually need to be connected to an energy network in order for them to work lol, I'll test your code and update the pr if it works (or if I figure something out that works). Ty for your feedback!